lib/models/views/profile.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:equatable/equatable.dart';
+import 'package:fusecash/models/app_state.dart';
+import 'package:redux/redux.dart';
+
+class ProfileViewModel extends Equatable {
+  final String phone;
+  final String Function() privateKey;
+  final String Function() firstName;
+
+  ProfileViewModel({this.phone, this.privateKey, this.firstName});
+
+  static ProfileViewModel fromStore(Store<AppState> store) {
+    return ProfileViewModel(
+        phone: store.state.userState.phoneNumber,
+        privateKey: () {
+          final key = store.state.userState.privateKey;
+          return '0x${key.substring(0, 2)}...${key.substring(key.length - 11)}';
```

Comment:
You need to show the walletAddress, not the privateKey.
PrivateKey can not be exposed!
also to format the walletAddress use the util function formatAddress, it's under utils/format.dat